phantom.py

```diff
@@ -0,0 +1,207 @@
+        scatterers[:, 1] = scatterers[:, 1] * (lims[3] - lims[2]) + lims[2]
+        scatterers[:, 2] = scatterers[:, 2] * (lims[5] - lims[4]) + lims[4]
+        self.scatterers = scatterers
+
+    def rigid_translate_scatterers(self):
+        """" Moves scatterers according to delta_xyz
+
+        """
+        self.translate_scatterers = np.random.random((self.n_scats, 3))
+        self.translate_scatterers[:, 0] = self.scatterers[:, 0] + (np.ones(self.n_scats) * self.delta_xyz[0])
+        self.translate_scatterers[:, 1] = self.scatterers[:, 1] + (np.ones(self.n_scats) * self.delta_xyz[1])
+        self.translate_scatterers[:, 2] = self.scatterers[:, 2] + (np.ones(self.n_scats) * self.delta_xyz[2])
+
+    def save_phantom(self):
+        """save phantom for specific timestep to HDF5 format file
+
+        """
+
+        h5out = h5py.File('{}.mat'.format(self.phantomout, self.timestep), 'a')
+        h5out.attrs()
+
+        h5out.create_group('timestep{03d}'.format(self.timestep))
+        h5out.attrs.__setitem__('delta_xyz', self.delta_xyz)
+        h5out.attrs.__setitem__('units', 'meters')
+
+        # What data should be stored in save_phantom? Positions and amplitudes? Come back to this later
+        # h5out.create_dataset('positions', data=phantom.positions, dtype='float')
+        # h5out.create_dataset('amplitudes', data=phantom.amplitudes, dtype='float')
+
+        h5out.close()
+
+    def make_dispdata(self):
+        """create displacement data in [dx, dy, dz] format
+
+        """
+        try:
+            from fem.post.create_disp_dat import parse_line
+        except ImportError:
+            print('ERROR: Problem importing fem.post.create_disp_dat.parse_line; check PYTHONPATH.')
+
+        nodout = open(self.nodout, "r")
+        n = nodout.readlines()
+        raw_data = []
+
+        # TODO: make dynamic text read in
+        for i in range(8, 18):
+            line = n[i]
+            raw_data.append(parse_line(line))
+
+        dispdata_complete = []
+
+        disp_iter = iter(raw_data)
+        for node in self.nodeIDcoords:
+            if any(item[0] for item in raw_data == node['id']):
+                thisdisp = next(disp_iter)
+                dispdata_complete.append([thisdisp[1], thisdisp[2], thisdisp[3]])
+            else:
+                dispdata_complete.append([0, 0, 0])
+
+        self.dispdat = dispdata_complete
+
+    def make_interp(self):
+        """3D interpolation
+
+        """
+
+        xinterp = np.interp([item[0] for item in self.scatterers], self.nodeIDcoords['x'], [item[0] for item in self.dispdat])
+        yinterp = np.interp([item[1] for item in self.scatterers], self.nodeIDcoords['y'], [item[1] for item in self.dispdat])
+        zinterp = np.interp([item[2] for item in self.scatterers], self.nodeIDcoords['x'], [item[2] for item in self.dispdat])
+
+        self.sdispdat = []
+        for i in range(len(xinterp)):
+            self.sdispdat.append([xinterp[i], yinterp[i], zinterp[i]])
```

Comment:
`append()` is a relatively "costly" approach to growing a list; could do more directly (and less iteratively) using a numpy array.

Reply:
fixed